### PR TITLE
fix NettyCodecAdapter decoder memory leak

### PR DESCRIPTION
**Pull request by JoeCqupt**
## What is the purpose of the change
**reproduce code**
```
    @Test
    void testDecodeException() throws IOException {
        Codec2 codec2 = Mockito.mock(Codec2.class);
        doThrow(new IOException("testDecodeIllegalPacket")).when(codec2).decode(any(), any());

        URL url = Mockito.mock(URL.class);
        doReturn("default").when(url).getParameter(eq(Constants.CODEC_KEY));

        ChannelHandler handler = Mockito.mock(ChannelHandler.class);
        NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, handler);
        io.netty.channel.ChannelHandler decoder = nettyCodecAdapter.getDecoder();
        EmbeddedChannel embeddedChannel = new EmbeddedChannel();
        embeddedChannel.pipeline().addLast(decoder);

        // simulate illegal data packet
        ByteBuf input = AbstractByteBufAllocator.DEFAULT.buffer();
        input.writeBytes("testDecodeIllegalPacket".getBytes(StandardCharsets.UTF_8));

        DecoderException decoderException = Assertions.assertThrows(DecoderException.class, () -> {
            embeddedChannel.writeInbound(input);
        });
        Assertions.assertTrue( decoderException.getCause() instanceof IOException);

        Assertions.assertEquals(0, input.refCnt());
    }
```

when NettyCodecAdapter decoder decode exception param ByteBuf not released

expect refcnt 0 , actual 1 

**Solution**
when decode excpetion . skip all readable bytes let `ByteToMessageDecoder` release ByteBuf

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [x] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by JoeCqupt** [comment]
@AlbumenJ  PTAL

File: dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java [hunk 1]
```diff
@@ -42,4 +55,30 @@ void test() {
...
+        URL url = Mockito.mock(URL.class);
+        doReturn("default").when(url).getParameter(eq(Constants.CODEC_KEY));
+
+        ChannelHandler handler = Mockito.mock(ChannelHandler.class);
+        NettyCodecAdapter nettyCodecAdapter = new NettyCodecAdapter(codec2, url, handler);
+        io.netty.channel.ChannelHandler decoder = nettyCodecAdapter.getDecoder();
+        EmbeddedChannel embeddedChannel = new EmbeddedChannel();
+        embeddedChannel.pipeline().addLast(decoder);
+
+        // 模拟问题数据包
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-netty4/src/test/java/org/apache/dubbo/remoting/transport/netty4/NettyCodecAdapterTest.java]: Comment in English

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — COMMENTED** [review]
@oxsean PTAL

**Review by oxsean — APPROVED** [review]
LGTM.

**Review by EarthChen — APPROVED** [review]
LGTM

**Pull request merged**